<?xml version="1.0"?>
<!--***********************************************************
	IHMC Interface Node
	Emily Sheetz, NSTGRO VTE 2021

	run either:
		$ roslaunch val_dynacore ihmc_interface_node.launch
		OR
		$ rosrun val_dynacore ihmc_interface_node
************************************************************-->
<launch>
	<arg name="controllers" default="true"/> <!-- indicates if joint commands come from controllers; will change queueing properties of IHMC messages -->
	<arg name="managing_node" default="ControllerTestNode"/> <!-- only necessary if controllers flag is true -->

	<!-- if not testing with controllers, robot pose and joint states will come from IKModuleTestNode -->
	<arg unless="$(arg controllers)" name="pelvis_tf_topic" value="/IKModuleTestNode/nstgro20_valkyrie_ik/pelvis_transform"/>
	<arg unless="$(arg controllers)" name="joint_command_topic" value="/IKModuleTestNode/nstgro20_valkyrie_ik/joint_commands"/>
	<!-- if testing with controllers, robot pose and joint states will come from managing_node -->
	<arg if="$(arg controllers)" name="pelvis_tf_topic" value="controllers/output/ihmc/pelvis_transform"/>
	<arg if="$(arg controllers)" name="joint_command_topic" value="controllers/output/ihmc/joint_commands"/>
	<arg if="$(arg controllers)" name="status_topic" value="controllers/output/ihmc/controller_status"/>
	<arg if="$(arg controllers)" name="controlled_link_topic" value="controllers/output/ihmc/controlled_link_ids"/>

	<arg name="debug" default="false"/>
	<arg unless="$(arg debug)" name="launch_prefix" value="" />
	<arg if="$(arg debug)" name="launch_prefix" value="xterm -e gdb **args" />

	<node launch-prefix="$(arg launch_prefix)" pkg="IHMCMsgInterface" type="ihmc_interface_node" name="IHMCInterfaceNode" output="screen">
		<param name="commands_from_controllers" value="$(arg controllers)"/>
		<param name="joint_command_topic" value="$(arg joint_command_topic)"/>
		<param name="pelvis_tf_topic" value="$(arg pelvis_tf_topic)"/>
		<!-- only controllers will send statuses and controlled links, otherwise status topic not needed -->
		<param if="$(arg controllers)" name="managing_node" value="$(arg managing_node)"/>
		<param if="$(arg controllers)" name="status_topic" value="$(arg status_topic)"/>
		<param if="$(arg controllers)" name="controlled_link_topic" value="$(arg controlled_link_topic)"/>
		<!--<param name="" type="" value=""/> -->
	</node>
</launch>
